sdk/src/test/scala/com/horizen/SidechainWalletTest.scala

```diff
@@ -627,4 +684,113 @@ class SidechainWalletTest
+    val fifthEpoch = ConsensusEpochNumber @@ 5
+    expectedEpochInfo = ConsensusEpochNumber @@ 3
+    assertTrue("No merkle path expected for unknown id.", sidechainWallet.getForgingBoxMerklePath(boxId, fifthEpoch).isEmpty)
+
+
+    // Test 4: get existent merkle path
+    assertTrue("Merkle path expected to be found.", sidechainWallet.getForgingBoxMerklePath(storedBoxId, fifthEpoch).isDefined)
+  }
+
+  @Test
+  def testApplyConsensusEpochInfo(): Unit = {
+    val mockedWalletBoxStorage: SidechainWalletBoxStorage = mock[SidechainWalletBoxStorage]
+    val mockedSecretStorage: SidechainSecretStorage = mock[SidechainSecretStorage]
+    val mockedWalletTransactionStorage: SidechainWalletTransactionStorage = mock[SidechainWalletTransactionStorage]
+    val mockedForgingBoxesMerklePathStorage: ForgingBoxesMerklePathStorage = mock[ForgingBoxesMerklePathStorage]
+    val mockedApplicationWallet: ApplicationWallet = mock[ApplicationWallet]
+
+    val sidechainWallet = new SidechainWallet(
+      "seed".getBytes(),
+      mockedWalletBoxStorage,
+      mockedSecretStorage,
+      mockedWalletTransactionStorage,
+      mockedForgingBoxesMerklePathStorage,
+      mockedApplicationWallet)
+
+
+    val forgerBox1 = getForgerBox
+    val forgerBox2 = getForgerBox
+    val forgerBox3 = getForgerBox
+
+    val forgerBox1ContainingTxId = bytesToId(forgerBox1.id())
+
+    val epochNumber: ConsensusEpochNumber = ConsensusEpochNumber @@ 3
+    val merkleTree: MerkleTree = MerkleTree.createMerkleTree(util.Arrays.asList(forgerBox1.id(), forgerBox2.id(), forgerBox3.id()))
+    val forgersStake: Long = forgerBox1.value() + forgerBox2.value() + forgerBox3.value()
+
+    val epochInfo: ConsensusEpochInfo = ConsensusEpochInfo(epochNumber, merkleTree, forgersStake)
+
+
+    Mockito.when(mockedWalletBoxStorage.getByType(ArgumentMatchers.any()))
+      .thenReturn(List(new WalletBox(forgerBox1, forgerBox1ContainingTxId, 1L)))
+
+    // Verify epoch number and return predefined merklePathSeq
+    Mockito.when(mockedForgingBoxesMerklePathStorage.update(
+      ArgumentMatchers.any[ConsensusEpochNumber](),
+      ArgumentMatchers.any[Seq[BoxMerklePathInfo]]()))
+      .thenAnswer(answer => {
+        val epoch = answer.getArgument(0).asInstanceOf[ConsensusEpochNumber]
+        val boxMerklePathInfoSeq = answer.getArgument(1).asInstanceOf[Seq[BoxMerklePathInfo]]
+
+        assertEquals("Different epoch number request expected.", epochNumber, epoch)
+        assertEquals("Different merkle path seq size expected.", 1, boxMerklePathInfoSeq.size)
+        assertArrayEquals("Different box id applied.", forgerBox1.id(), boxMerklePathInfoSeq.head.boxId)
+        assertArrayEquals("Wrong merkle path applied.", merkleTree.rootHash(), boxMerklePathInfoSeq.head.merklePath.apply(forgerBox1.id()))
+
+        Success(mockedForgingBoxesMerklePathStorage)
+      })
+
+    sidechainWallet.applyConsensusEpochInfo(epochInfo)
```

Comment:
check of stored consensus info information?

Reply:
ok

Reply:
found. it was done in testGetForgingBoxMerklePath. Check of what was applied to storage was done in line 779.